qiskit/providers/models/backendproperties.py

```diff
@@ -122,8 +126,156 @@ def __init__(self, backend_name, backend_version, last_update_date,
+            for prop in props:
+                value = self._apply_prefix(prop.value, prop.unit)
+                formatted_props[prop.name] = (value, prop.date)
+                self._qubits[qubit] = formatted_props
+
+        self._gates = {}
+        _temp = {}  # To add nested dictionaries
+        _temp_name = str  # To check for list of dictionaries
+        for gate in gates:
+            qubits = _to_tuple(gate.qubits)
+            formatted_props = {}
+            formatted_props['name'] = gate.name
+            for param in gate.parameters:
+                value = self._apply_prefix(param.value, param.unit)
+                formatted_props[param.name] = (value, param.date)
+            if _temp_name != gate.gate:
+                _temp = {}
+            _temp_name = gate.gate
+            _temp[qubits] = formatted_props
+            self._gates[gate.gate] = _temp
 
         super().__init__(**kwargs)
+
+    def gate_error(self, operation: str, qubits: Union[int, Iterable[int]]) -> float:
+        """
+        Return gate error estimates from backend properties.
+
+        Args:
+            operation (str): The operation for which to get the error.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Return:
+            'Gate error' of the given operation and qubit(s).
+
+        Raises:
+            PulseError: If error is True and the property is not found.
+        """
+        return self.gate_property(operation, qubits,
+                                  'gate_error')[0]  # Throw away datetime at index 1
+
+    def gate_length(self, operation: str, qubits: Union[int, Iterable[int]]) -> float:
+        """
+        Return the duration of the gate in units of seconds.
+
+        Args:
+            operation (str): The operation for which to get the duration.
+            qubits (Union[int, Iterable[int]]): The specific qubits for the operation.
+
+        Return:
+            'Gate length' of the given operation and qubit(s).
+
+        Raises:
+	        PulseError: If error is True and the property is not found.
+        """
+        return self.gate_property(operation, qubits,
+                                  'gate_length')[0]  # Throw away datetime at index 1
+
+    def gate_property(self,
+                      operation: str,
+                      qubits: Union[int, Iterable[int]] = None,
+                      name: str = None) -> Tuple[Any, datetime.datetime]:
+        """
+        Return the gate properties of the given qubit and property.
+
+        Args:
+            operation (str) : Name of the gate
+            qubits (Union[int, Iterable[int]]): The qubit to find properties for.
+            name (str): Optionally used to specify within the heirarchy which
+                        property to return.
+
+        Return:
+            Gate properties of the given qubit and property, if it was given by the
+            backend, otherwise, return `None`.
```

Comment:
This isn't true actually, if the property isn't found, this will error. I'm not sure if returning None or an error is preferable (that is why I had an `error` argument at some point)